crates/relay/src/lib.rs

```diff
@@ -261,6 +344,23 @@ impl<T: Config> Pallet<T> {
         Ok(())
     }
 
+    pub(crate) fn has_input_from_wallet(transaction: &Transaction, wallet: &Wallet) -> bool {
+        // collect all addresses that feature in the inputs of the transaction
+        let input_addresses: Vec<Result<BtcAddress, _>> = transaction
+            .clone()
+            .inputs
+            .into_iter()
+            .map(|input| input.extract_address())
+            .collect();
+
+        // TODO: can a vault steal funds if it registers a P2WPKH-P2SH since we
```

Comment:
Did you make a ticket?

Reply:
This was noted in the original audit, but nothing was identified so I think we can remove. cc: @shonfeder